### PR TITLE
Add design docs on taints and tolerations, gang scheduling

### PR DESCRIPTION
**Pull request by islinwb**

### CODE REVIEW

File: docs/design/taints_and_tolerations_design.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (m1093782566) [inline docs/design/taints_and_tolerations_design.md]: please remove the extra new lines.

**Comment by m1093782566** [comment]
cc @jiaxuanzhou

**Comment by jiaxuanzhou** [comment]
looks good, but one suggestion:
the proposal does not mention the priority of the jobs, here is one scenario:
job1 (high priority), job2(low priority)
pods of job0 are running, pods of job1/job2 are pending as there is not enough resource,
when job0 completed, resource is released and can only satisfy  the request of job1 or job2, but only  pods of job1 can be placed as it has high priority.

would you also consider this feature in your design?

**Comment by m1093782566** [comment]
cc @deepak-vij @shivramsrivastava for inputs.

File: docs/design/taints_and_tolerations_design.md (hunk 1 above)
Comment (islinwb) [reply docs/design/taints_and_tolerations_design.md]: done

**Comment by m1093782566** [comment]
I think we can let these two design proposals in for now. After that, we can refine the gang scheduling design document per the codes.

**Pull request merged**